snuba/utils/streams/synchronized.py

```diff
@@ -43,7 +71,7 @@ class SynchronizedConsumer(Consumer[TPayload]):
     def __init__(
         self,
         consumer: Consumer[TPayload],
```

Comment:
If the commit_log_consumer is coupled with KafkaPayload, does it make sense coupling the consumer as well ?
Or is there a reason to keep `consumer` still decoupled from the design standpoint ?

Reply:
I'm hoping the coupling is a temporary thing.

I mostly didn't implement the changes described [in this comment ](https://github.com/getsentry/snuba/pull/1060/files/2d49695f92b36507f62de3ea8de42638bea05365#r447227047) because there's going to be a lot of stuff moving around over the next couple of weeks and I didn't want to increase the potential surface area for any future changes I haven't already anticipated and it's a lot of boilerplate to do it the codec consumer/producer "right" way. Ideally the consumer here goes back to `Consumer[Commit]` soon, and I very well may find myself in a spot soon enough that motivates me to do address this in the near future if things keep going the way they've been going.

The biggest reason to avoid to making `Consumer` (big-C) bound on `KafkaPayload` unilaterally is that means the subscription consumer that expects `Consumer[Tick]` would have to change. I don't foresee subscriptions needing to change to use the multiprocessing architecture at the time being, but I suppose that could change in the future too.